sheets/quickstart/quickstart.py

```diff
@@ -29,15 +30,25 @@ def main():
     """Shows basic usage of the Sheets API.
     Prints values from a sample spreadsheet.
     """
-    # The file token.json stores the user's access and refresh tokens, and is
-    # created automatically when the authorization flow completes for the first
-    # time.
-    store = file.Storage('token.json')
-    creds = store.get()
-    if not creds or creds.invalid:
-        flow = client.flow_from_clientsecrets('credentials.json', SCOPES)
-        creds = tools.run_flow(flow, store)
-    service = build('sheets', 'v4', http=creds.authorize(Http()))
+    creds = None
+    # See if there is a token.dat from a previous run
+    # that contains the required credentials.
+    if os.path.exists('token.dat'):
+        with open('token.dat', 'rb') as token:
+            creds = pickle.load(token)
+        # The token expires after an hour and may need to be refreshed.
+        if creds.expired:
+            creds.refresh(Request())
```

Comment:
You need to check if `creds.refresh_token` exists before attempting a refresh. For various reasons it is possible to get an access token without a refresh token, and in the case where that has happened we should throw the user back into the `InstalledAppFlow`.